client/components/data/domain-management/email/index.jsx

```diff
@@ -48,84 +51,125 @@ function getStateFromStores( props ) {
 	};
 }
 
-const EmailData = React.createClass( {
-	displayName: 'EmailData',
+export class EmailData extends Component {
+	static propTypes = {
+		context: PropTypes.object.isRequired,
+		fetchSitePlans: PropTypes.func.isRequired,
+		products: PropTypes.object.isRequired,
+		selectedDomainName: PropTypes.string,
+		selectedSite: PropTypes.object.isRequired,
+		sitePlans: PropTypes.object.isRequired,
+		googleAppsUsers: PropTypes.array.isRequired,
+		googleAppsUsersLoaded: PropTypes.bool.isRequired,
+	};
 
-	propTypes: {
-		component: React.PropTypes.func.isRequired,
-		context: React.PropTypes.object.isRequired,
-		productsList: React.PropTypes.object.isRequired,
-		selectedDomainName: React.PropTypes.string,
-		sitePlans: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired,
-		googleAppsUsers: React.PropTypes.array.isRequired,
-		googleAppsUsersLoaded: React.PropTypes.bool.isRequired
-	},
+	constructor( props ) {
+		super( props );
 
-	mixins: [ observe( 'productsList' ) ],
+		props.loadDomainsAndSitePlans();
```

Comment:
I see for the first time a network call in the ES6 class constructor call. I couldn't find anything about related common practices in React docs (https://facebook.github.io/react/docs/reusable-components.html#es6-classes). I'm just leaving a comment to make sure we double check if it's okey or rather we should use `componentDidMount` method to do it. It looks like constructor's role is to set initial state and explicitly bind handlers to `this`.